### ISSUE TITLE
When adding row with widths :auto and styling of textRotation: 90, width incorrect

### ISSUE THREAD

**Issue by untoldone** [caxlsx/caxlsx#49]
It appears that autowidth is calculating a width assuming the content of the cell is horizontally written.

Today, if you set text rotation for a cell in the column that has the widest content in the column, autowidth seems to make the cell as wide as that content, despite it being narrow/ tall content given the text has been rotated.

My expectation is that autowidth would set the width to be as wide as the content visually, where as today it seems to be setting the width to be the width of the text assuming it is written horizontally.

Reproduction:

```
require "axlsx"

p = Axlsx::Package.new
wb = p.workbook
cell_rotated_text_style = wb.styles.add_style(alignment: {textRotation: 90})

wb.add_worksheet(name: "Testing") do |sheet|
  sheet.add_row ["hello world this is some long text"], {style: cell_rotated_text_style}
  sheet.add_row ["12"], widths: [:auto]
end
p.serialize("test.xlsx")
```

**Comment by noniq** [comment]
You're right, autowidth calculation currently does not take text rotation into account.

Unfortunately the whole autowidth-related code is rather “hacky”, see https://github.com/caxlsx/caxlsx/pull/44 for a recent attempt of improving it a bit (it finally succeeded, but was way more complicated than initially estimated).

Nonetheless, if you want to give it a try: We’d be happy to see a pull request!

**Comment by untoldone** [comment]
Just a thought -- should this be documented as a known issue for folks in the future before being closed? I'm ok either way as I clearly haven't contributed a fix + I understand there's limited time.
